### DISCUSSION TITLE
Provide public api for audit functionality

### DISCUSSION THREAD

**Discussion by LordSimal** [composer/composer#11110] [General]
Currently the `Auditor` class is marked as `@internal` and therefore (I would guess) discouraged to be used directly outside of the composer package.

But it would be awesome if some sort of public API would be available to perform audit checks programatically via something like
```
$auditor = new \Composer\Advisory\Auditor();
$result = $auditor->auditFiles($composerJson, $composerLock);
```

Where `$composerJson` is a string representation of the `composer.json` and `$composerLock` a string representation of the `composer.lock`

Maybe something similar like it is explained [HERE](https://getcomposer.org/doc/07-runtime.md) with e.g.
```
\Composer\InstalledVersions::getVersion('vendor/package');
```

**Comment by LordSimal** [comment]
Actually I just discovered that `composer audit` ONLY works if you actually have the modules installed via `composer install`
Just having the `composer.json` and `composer.lock` files placed inside a folder doesn't actually result in a audit being returned.

**Reply by stof** [reply]
By default, `composer audit` is auditing the vendor folder. there is a `--locked` flag if you want to audit based on the lock file instead.

**Reply by LordSimal** [reply]
Ah this is nice! Thank you! This definitely is easier for me since I now only have to download those 2 files inside a folder and execute the command on it 👍🏻 
But still I would prefer some more "direct" class approach instead of having to do a `shell_exec` or a 
```
new Process(['composer', 'audit', '--format=json', '--no-interaction', '--locked']);
```
